Sources/NIOSSL/SSLContext.swift

```diff
@@ -569,6 +567,39 @@ extension NIOSSLContext {
             parentSwiftContext.keyLogManager!.log(linePointer)
         }
     }
+    
+    /// Takes a path and determines if the file at this path is of c_rehash format .
+    internal static func _isRehashFormat(path: String) throws -> Bool {
+        // Check if the element’s name matches the c_rehash symlink name format.
+        // Note that `lastPathComponent` is not available here.
+        guard let lastPathComponent = path.split(separator: "/").last else { return false }
```

Comment:
So it's up to you whether you want to take this, but an optional refactor here would be to do all of this computation on the `UTF8View` of `path`. That is, do `let utf8View = path.utf8` at the top and then do all of the rest of the `split` and similar operations on the subsequences of the `UTF8View`.

Aside from a few changes to `UInt8(ascii:)` instead of string separators this should mostly be equivalent. We can also delay constructing the `filename` string until just before the call to `lstat`.

In fact, that doesn't even need to be a string: you could do:

```swift
let filename = filenameElement.map { CInt(bitPattern: $0) }
```

It's a bit cheaper to get the pointer out of Array than out of String so that'll help a bit.

Totally up to you though, and we can always do it in a follow-up PR.

Reply:
Thanks @Lukasa.  I'll take an additional look here to include this refactor.  As a side note, I believe you also mentioned this earlier: 
> For performance reasons we should probably rewrite all of this code to work on String.UTF8View

And I had thought you were referring to only this snippet of code:
```swift
guard fileExtension != nil,
              filename.count == 8,
            filename.allSatisfy(\.isHexDigit) else { return false }
```